### PR TITLE
[core] Fix the test_failure_3.py in win

### LINKED ISSUE #26697: [core] test_failure_3.py failed on windows.
### What happened + What you expected to happen

When GIL is released, windows test starts to fail on the newly added tests. We need to understand why.

https://github.com/ray-project/ray/pull/26696

### Versions / Dependencies

master

### Reproduction script

https://github.com/ray-project/ray/pull/26696

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
Win tests were broken because when the child is killed, the parent is also killed. Change the signal sent and make it work.

## Related issue number

Closes #26697 
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_failure_3.py [hunk 1]
```diff
@@ -155,7 +154,7 @@ def pid(self):
     rs = []
 
     def submit():
-        for i in range(100000):
+        for i in range(10000):
```
Comment (rkooo567) [inline python/ray/tests/test_failure_3.py:157]: should we keep this?

**Review by rkooo567 — APPROVED** [review]
OOC, but what's the difference between signal.SIGKILL and SIGKILL? lol

File: python/ray/tests/test_failure_3.py (hunk 1 above)
Comment (iycheng) [reply python/ray/tests/test_failure_3.py:157]: I tested it and I notice it's going to run for such a long time before it terminated. So I reduce the size.

**Comment by iycheng** [comment]
@rkooo567 it's define in L16 in this test:

`SIGKILL = signal.SIGKILL if sys.platform != "win32" else signal.SIGTERM`

I guess that's because in windows there is no SIGKILL?

**Pull request merged**